src/foam/dao/SequenceNumberDAO.java

```diff
@@ -4,15 +4,25 @@
 import foam.core.FObject;
 import foam.core.X;
 import foam.core.PropertyInfo;
+import foam.mlang.MLang;
 import foam.mlang.sink.Max;
 
 public class SequenceNumberDAO
-  extends ProxyDAO
+    extends ProxyDAO
 {
   protected String property = "id";
-  protected int value = 1;
+  protected long value = 1;
   private PropertyInfo property_ = null;
 
+  public SequenceNumberDAO(DAO delegate) {
+    setDelegate(delegate);
+  }
+
+  public SequenceNumberDAO(DAO delegate, String property) {
```

Comment:
Less error prone if you let the user specify using a PropertyInfo rather than a String. Then, if the property is removed they'll get a compile-time error.